verify_test.go

```diff
@@ -182,6 +183,126 @@ JBOFyaCnMotGNioSHY5hAkEAxyXcNixQ2RpLXJTQZtwnbk0XDcbgB+fBgXnv/4f3
 BCvcu85DqJeJyQv44Oe1qsXEX9BfcQIOVaoep35RPlKi9g==
 -----END PRIVATE KEY-----`
 
+func TestVerifyWithHashCalcFunc(t *testing.T) {
+	fixture := UnmarshalTestFixture(HashCalcSignedTestFixture)
+	p7, err := Parse(fixture.Input)
+	if err != nil {
+		t.Errorf("Parse encountered unexpected error: %v", err)
+	}
+
+	const longBuffer = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam a molestie odio, id accumsan dolor. Praesent ultricies enim et pharetra molestie. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur vitae pellentesque tortor. Curabitur nulla mi, semper non lectus nec, auctor euismod tellus. Nunc vestibulum nisi quis felis efficitur, vel finibus nunc vehicula. Mauris ipsum mi, eleifend in urna non, pellentesque facilisis turpis. Ut eleifend viverra imperdiet. Vestibulum ut ligula non nunc vestibulum lobortis. Curabitur at elementum nisl. Sed facilisis ligula in pulvinar aliquet. Sed semper interdum ipsum quis hendrerit."
+	reader := bytes.NewReader([]byte(longBuffer))
+	p7.HashCalc = func(h hash.Hash, b []byte) []byte {
+		bufferSize := 128
+		buffer := make([]byte, bufferSize)
+		for {
+			count, err := reader.Read(buffer)
+			if err != nil {
+				break
+			}
+			h.Write(buffer[:count])
+		}
+		return h.Sum(nil)
```

Comment:
This is just an example for the test, but in an actual implementation you need something like this:

```suggestion
		for {
			count, err := reader.Read(buffer)
			if count > 0 {
			        h.Write(buffer[:count])
		        }
		        if err == io.EOF {
			        break
		        }
		        if err != nil {
			    return nil, err
		        }
		}
		
		return h.Sum(nil)
```